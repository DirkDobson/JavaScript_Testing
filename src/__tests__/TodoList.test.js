import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TodoList from '../TodoList'


describe('<TodoList />', () => {
  describe('render', () => {
  let tree

  beforeEach( () => {
    let component = shallow(<TodoList />)
    tree = toJson(component)
  })
  it('renders without crashing', () => {
  shallow(<TodoList />)
})

it('matches a snapshot', () => {
expect(tree).toMatchSnapshot()
})
})

describe('functionality', () => {
  let component 

  beforeEach( () => {
    component = shallow(<TodoList />)
  })

  it('updates state on change', () =>{
    const input = component.find('input')
    input.simulate('focus')
    input.simulate('change', { target: {name: 'name', value: 'Hello'}})
    expect(component.state('name')).toEqual('Hello')
  })
})
})
