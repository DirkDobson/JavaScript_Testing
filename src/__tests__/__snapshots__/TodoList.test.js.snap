// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> render matches a snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <label
      style={Object {}}
    >
      Add a Todo
    </label>
    <input
      name="name"
      onChange={[Function]}
      required={true}
      value=""
    />
    <ul />
  </form>
</Fragment>
`;
